packages/aft/lib/src/commands/review_command.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:aft/aft.dart';
+
+class GitHubData {
+  GitHubData.fromJson(Map<String, dynamic> json)
+      : title = json['title'] as String,
+        body = json['body'] as String,
+        number = json['number'] as int,
+        changedFiles = json['changedFiles'] as int,
+        additions = json['additions'] as int,
+        deletions = json['deletions'] as int,
+        url = json['url'] as String;
+
+  String title;
+  String body;
+  int number;
+  int changedFiles;
+  int additions;
+  int deletions;
+  String url;
+
+  final String _legal =
+      'By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.';
+  final String _lineBreak = '========================================';
+
+  final emojis = {
+    ReviewPriority.urgent: ':alert:',
+    ReviewPriority.high: ':arrow_double_up:',
+    ReviewPriority.normal: ':arrow_up:',
+    ReviewPriority.low: ':arrow_down:',
+  };
+
+  @override
+  String toString() {
+    return '{Title: $title,\nBody: $body,\nNumber: $number,\nChanged Files: $changedFiles,\nAdditions: $additions,\nDeletions: $deletions}';
+  }
+
+  /// Converts the PR data to a Slack-formatted string.
+  String toMarkdown(ReviewPriority priority) {
+    return '''
+$_lineBreak
+${emojis[priority]} *$title - [$number]($url)*
+> +$additions -$deletions | files: $changedFiles
+${body.replaceAll(_legal, '').trim()}
+*Reviewers:*  
+$_lineBreak''';
+  }
+}
+
+enum ReviewPriority { urgent, high, normal, low }
+
+/// Command to clean all temporary files in the repo.
+class ReviewCommand extends AmplifyCommand {
+  ReviewCommand() {
+    argParser
+      ..addOption(
+        'ref',
+        abbr: 'r',
+        help:
+            'The PR reference. Can be a PR number or URL. Defaults to the current branch.',
+      )
+      ..addOption(
+        'priority',
+        abbr: 'p',
+        help: 'Determines the emoji to use for the PR review.',
+        allowed: ReviewPriority.values.map((e) => e.name).toList(),
+        defaultsTo: ReviewPriority.normal.name,
+      );
+  }
+  @override
+  String get description => 'Creates PR review text for slack';
+
+  @override
+  String get name => 'review';
+
+  /// The PR ref to review.
+  late final ref = argResults?['ref'] as String? ?? '';
+
+  /// Determines the emoji to use for the PR review.
+  late final priority = ReviewPriority.values.firstWhere(
+    (e) => e.name == argResults!['priority'] as String,
+    orElse: () => ReviewPriority.normal,
+  );
+
+  /// Verifies that the PR is ready using gh CLI.
+  Future<void> _verifyPR(String ref) async {
+    final branchName = await Process.run(
+      'gh',
+      ['pr', 'ready', ref],
+    );
+    if (branchName.exitCode != 0) {
+      logger.error('A PR is not ready at ref: $ref');
```

Comment:
*suggestion:* Check/log `stderr` since the info can be useful. For example if there PR is closed the error message is `X Pull request #4793 is closed. Only draft pull requests can be marked as "ready for review"` which is useful info to log.

```suggestion
      final error = branchName.stderr.toString();
      if (error.contains('Could not resolve to a PullRequest')) {
        logger.error('A PR is not ready at ref: $ref');
      } else {
        logger.error('Invalid ref: $error');
      } 
```